libraries/botbuilder-ai/src/qnaMakerDialog.ts

```diff
@@ -299,9 +377,18 @@ export class QnAMakerDialog extends WaterfallDialog implements QnAMakerDialogCon
         if (top) {
             this.top = new IntExpression(top);
         }
-        if (activeLearningCardTitle) {
-            this.activeLearningCardTitle = new StringExpression(this.defaultCardTitle);
+        if (isSuggestionsFactory(activeLearningTitleOrFactory)) {
+            if (!cardNoMatchText) {
+                // Without a developer-provided cardNoMatchText, the end user will not be able to tell the convey to the bot and QnA Maker that the suggested alternative questions were not correct.
+                // When the user's reply to a suggested alternatives Activity matches the cardNoMatchText, the QnAMakerDialog sends this information to the QnA Maker service for active learning.
+                throw new TypeError('cardNoMatchText is required when using the suggestionsActivityFactory.');
```

Comment:
It's a bit pedantic, but this isn't really a type error. MDN defines a type error as `an error when an operation could not be performed, typically (but not exclusively) when a value is not of the expected type`. This is business logic that encodes an invalid combination of parameters as an error, not the wrong type.